lib/easy_image_viewer.dart

```diff
@@ -107,22 +132,10 @@ Future<Dialog?> showImageViewerPager(
                         onPressed: () {
                           Navigator.of(context).pop();
 
-                          if (onViewerDismissed != null) {
-                            onViewerDismissed(
-                                pageController.page?.round() ?? 0);
-                          }
-
-                          if (immersive) {
-                            SystemChrome.setEnabledSystemUIMode(
-                                SystemUiMode.edgeToEdge);
-                          }
-                          if (internalPageChangeListener != null) {
-                            pageController
-                                .removeListener(internalPageChangeListener);
-                          }
-                          pageController.dispose();
+                          handleDismissal();
```

Comment:
Question: Will the `Navigator.of(context).pop();` above trigger the `WillPopScope`'s `onWillPop` to fire, causing this to be redundant? Or will `onWillPop` not get triggered by this `.pop()`?

Reply:
@justinlettau Excellent question. `Navigator.pop` does not call `onWillPop`, see https://github.com/flutter/flutter/issues/31054#issuecomment-623713030. Also, I put a breakpoint into `handleDismissal` and it's only called once.